kong/db/schema/others/declarative_config.lua

```diff
@@ -955,10 +1030,33 @@ function DeclarativeConfig.load(plugin_set, vault_set, include_foreign)
     return nil, err
   end
 
-  -- we replace the "foreign"-type fields at the top-level
-  -- with "string"-type fields only after the subschemas have been loaded,
-  -- otherwise they will detect the mismatch.
+  -- Pre-load the full schema to validate the schema for sync.v2. Lazy loading
+  -- the full schema with DeclarativeConfig.load() will consume a lot of time
+  -- due to load_plugin_subschemas().
+  local full_schema
+
+  if sync_v2_enabled then
```

Comment:
Could we move this block to line 1077?

Reply:
i think it could not work to move.

see this comment:

<img width="472" alt="image" src="https://github.com/user-attachments/assets/79faca3c-c804-482c-9ec8-bca2ffcfff91" />

so if you move there, the foreignt-type fields will be replaced with string-type